src/store/datasets.js

```diff
@@ -142,7 +170,23 @@ export const makeActions = (dependencies) => ({
    * @param {File[]} files
    */
   async loadFiles({ dispatch }, files) {
+    // Recurse to load any compressed files
+    const zips = files.filter((f) => getExtension(f.name) === 'zip');
     const { fileIO } = dependencies;
+    const supportedExts = ['zip']
+      .concat(Object.keys(fileIO.fileReaders))
+      .concat(Object.keys(fileIO.typeAliases));
+    if (zips.length) {
+      const nonzips = files.filter((f) => getExtension(f.name) !== 'zip');
+      const p = zips.map((file) =>
+        JSZip.loadAsync(file).then((zip) =>
+          Promise.all(zipGetSupportedFiles(zip, null, supportedExts))
```

Comment:
We will have to re-think how we get the "supported files". The FileIO object uses two techniques to determine if a file is supported: file extension, and file magic. There are times when DICOM files have no extension, yet should still be processed as DICOM.

I suspect we will have to load every file in a zip and use fileIO to test to see if they are loadable. It's not performant compared to just a name extension check, but I'll assume that people loading zips only have images inside of them.

Reply:
Let's change this to first extract all files out of the zip, and them pass them through the loader. This way we can process DICOM files that don't necessarily have the .dcm extension.